### PR TITLE
chore: update sanity suite

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

- Source config test data
- Fix comparing source config test data ignoring the order of the destinations
- Add "View diff" option for viewing payload differences for failed test cases.
  - This functionality doesn't reliably work for legacy browsers that do not support longer URLs.

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-1757/enhance-sanity-suite-js-sdk

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [x] Chrome
- [ ] Firefox
- [x] IE11

## Sanity Suite

- [x] All sanity suite test cases pass locally

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Enhanced the test results display in the TestBook module, now showing both actual and expected results in a structured JSON format.
	- Added base64 encoding and decoding functionality for string handling in tests.

- **Enhancements**
	- Improved the user interface for test cases in TestBook, including updated HTML elements and styles for better readability and interaction.
	- Refined the logic for result comparison in tests, offering clearer differentiation between expected and actual outcomes.

- **Bug Fixes**
	- Updated configurations to include new properties for live event timestamp tracking across different source destinations.

### CODE REVIEW

**Review by MoumitaM — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -262,7 +265,7 @@ rudderanalytics.ready(() => {
 ## [](https://github.com/rudderlabs/rudder-sdk-js/blob/main/README.md#self-hosted-control-plane)Self-hosted control plane
 
 | **[Control Plane Lite](https://www.rudderstack.com/docs/get-started/rudderstack-open-source/control-plane-lite/) is now deprecated. It will not work with the latest rudder-server versions (after v1.2). Using [RudderStack Open Source](https://www.rudderstack.com/docs/get-started/rudderstack-open-source/control-plane-setup/#rudderstack-open-source) to set up your control plane is strongly recommended.** |
-| :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md:268]: The committable suggestion is not valid.

File: packages/sanity-suite/src/testBook/string.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Converts a base64 encoded string to bytes array
+ * @param base64Str base64 encoded string
+ * @returns bytes array
+ */
+const base64ToBytes = base64Str => {
+  const binString = globalThis.atob(base64Str);
+  const bytes = binString.split('').map(char => char.charCodeAt(0));
+  return new Uint8Array(bytes);
+};
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/sanity-suite/src/testBook/string.js]: The function has been removed now.

File: packages/sanity-suite/src/testBook/string.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  const binString = Array.from(bytes, x => String.fromCodePoint(x)).join('');
+  return globalThis.btoa(binString);
+};
+
+/**
+ * Encodes a string to base64 even with unicode characters
+ * @param value input string
+ * @returns base64 encoded string
+ */
+const toBase64 = value => bytesToBase64(new TextEncoder().encode(value));
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/sanity-suite/src/testBook/string.js]: Addressed.

File: packages/sanity-suite/src/testBook/string.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Decodes a base64 encoded string
+ * @param value base64 encoded string
+ * @returns decoded string
+ */
+const fromBase64 = value => new TextDecoder().decode(base64ToBytes(value));
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/sanity-suite/src/testBook/string.js]: The function has been removed now.

File: packages/sanity-suite/src/testBook/ResultAssertions.js [hunk 5]
```diff
@@ -46,7 +46,10 @@ const ResultsAssertions = {
         );
       }
 
-      return JSON.stringify(resultData, undefined, 2);
+      return {
+        resultData: JSON.stringify(resultData, undefined, 2),
+        expectedResultData: JSON.stringify(expectedResultData, undefined, 2),
+      };
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/sanity-suite/src/testBook/ResultAssertions.js:52]: Addressed.

File: README.md [hunk 6]
```diff
@@ -196,23 +196,25 @@ A sample `identify` call is shown below:
...
+      search: 'estore bestseller',
+      title: 'The best sellers offered by EStore',
+      url: 'https://www.estore.com/best-seller/1',
+    },
   },
   () => {
-    console.log("identify call");
-  }
+    console.log('identify call');
+  },
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md]: Addressed.

**Pull request merged**